frontend/app/routes/_gcweb-app.personal-information.home-address.edit.tsx

```diff
@@ -73,17 +74,37 @@ export async function action({ request }: ActionFunctionArgs) {
   const formData = Object.fromEntries(await request.formData());
   const parsedDataResult = await formDataSchema.safeParseAsync(formData);
 
+  const wsaddressWebService = await getWSAddressService();
```

Comment:
**Suggestion:** Move this line below the `if (!parsedDataResult.success)` block for the following reasons:
- We won't unnecessarily create the `WSAddressService` if form validation failed
- We keep initialization of variables close to where they are being used (`wsAddressWebService` is initialized on this line but is used seven lines later).

**Nitpick:** Consider keeping the name of the variable consistent with the service name
```suggestion
  const wsAddressService = await getWSAddressService();
```

Reply:
done